### PR TITLE
Fix u16int_t time overflow bug

### LINKED ISSUE #2: Stops working on SAMD devices (Arduino Zero ...)
Due to the different architecture (AVR: 8 bit, ARM 32 bit) there is an overflow condition in the PD_UFP_c timer function.
Any compare between an arithmetic expression needs to be casted to uint16_t. If not the alert function will not be called after 65536ms

like so:

bool PD_UFP_c::timer(void)
{
    uint16_t t = clock_ms();
    if (wait_src_cap && ((uint16_t)(t - time_wait_src_cap) > t_TypeCSinkWaitCap)) {
        time_wait_src_cap = t;
        if (get_src_cap_retry_count < 3) {
            uint16_t header;
            get_src_cap_retry_count += 1;
            /* Try to request soruce capabilities message (will not cause power cycle VBUS) */
            PD_protocol_create_get_src_cap(&protocol, &header);
            status_log_event(STATUS_LOG_MSG_TX);
            FUSB302_tx_sop(&FUSB302, header, 0);
        } else {
            get_src_cap_retry_count = 0;
            /* Hard reset will cause the source power cycle VBUS. */
            FUSB302_tx_hard_reset(&FUSB302);
            PD_protocol_reset(&protocol);
        }
    }
    if (wait_ps_rdy) {
        if ((uint16_t)(t - time_wait_ps_rdy) > t_RequestToPSReady) {
            wait_ps_rdy = 0;
            set_default_power();
        }
    } else if (send_request || (status_power == STATUS_POWER_PPS && ((uint16_t)(t - time_PPS_request) > t_PPSRequest))) {
        wait_ps_rdy = 1;
        send_request = 0;
        time_PPS_request = t;
        uint16_t header;
        uint32_t obj[7];
        /* Send request if option updated or regularly in PPS mode to keep power alive */
        PD_protocol_create_request(&protocol, &header, obj);
        status_log_event(STATUS_LOG_MSG_TX, obj);
        time_wait_ps_rdy = clock_ms();
        FUSB302_tx_sop(&FUSB302, header, obj);
    }
    if ((uint16_t)(t - time_polling) > t_PD_POLLING) {
        time_polling = t;
        return true;
    }
    return false;
}

### PR DESCRIPTION
**Pull request by jarrod89**
Fixes #2 for 32bit uCs (SAMD, ESP32 etc)
TLDR: This bug broke PPS operation on those uCs

### CODE REVIEW

**Review by kcl93 — APPROVED** [review]
@jarrod89 thanks for the fixes. To be honest, I never got to fully finish work on this project due to some private changes in my recent life. I am positively surprised that it actually works and hope, that I will do it fine for you :)
